src/app/chat/chat.component.ts

```diff
@@ -27,34 +27,60 @@ export class ChatComponent {
     userId: string | null;
     username: string | null;
   }> = this.store.select(selectUserData);
-  protected readonly datasource: IDatasource<Message> = this.getDatasource();
+
+  protected readonly datasource = new Datasource<Message>({
+    get: (index: number, count: number) => {
+      console.log(index, count);
+      return this.store
+        .select(selectMessagesSlice(index, count))
+        .pipe(take(1), tap(console.log));
+    },
+    devSettings: { debug: true },
+  });
```

Comment:
Since the update is not total, the Datasource should remain. But we need to provide merge logic, see below.